res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,714 @@
+        }
+    }
+};
+
+ReloopBeatmix24.ActivateFx2 = function(channel, control, value, status, group) {
+    if (value === DOWN) {
+        if (FxModeTimers[group]) {
+            engine.stopTimer(FxModeTimers[group]);
+            delete FxModeTimers[group];
+        }
+        FxModeLongPressed[group] = false;
+        FxModeTimers[group] = engine.beginTimer(1000,
+            "ReloopBeatmix24.FxModeCallback(\"" + group + "\", 2)",
+            true);
+    } else { // UP
+        if (FxModeLongPressed[group]) { // long press
+            // Nothing to do, this has already been done in callback function
+            FxModeLongPressed[group] = false;
+        } else { // short press
+            // stop & delete timer
+            engine.stopTimer(FxModeTimers[group]);
+            delete FxModeTimers[group];
+            var oldvalue = engine.getValue("[EffectRack1_EffectUnit2]",
+                "group_" + group + "_enable");
+            engine.setValue("[EffectRack1_EffectUnit2]", "group_" + group +
+                "_enable", oldvalue ? 0 : 1);
+        }
+    }
+};
+
+ReloopBeatmix24.FxKnobTurn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        var parameter = control;
+        engine.setParameter(group, "parameter" + parameter.toString(),
+            script.absoluteLin(value, 0, 1));
+    }
+    // Nothing in multi-effect mode
+};
+
+ReloopBeatmix24.ShiftFxKnobTurn = function(channel, control, value, status,
+    group) {
+    if (FxMode == 1) {
+        var parameter = 3 + control - SHIFT;
+        engine.setParameter(group, "parameter" + parameter.toString(),
+            script.absoluteLin(value, 0, 1));
+    } else {
+        var effectUnit = parseInt(group.substr(23, 1), 10);
+        var Effect = control - SHIFT;
+        var storeIndex = "FX" + Effect.toString() + "U" + effectUnit.toString();
+        if (storeIndex in previousValue) {
+            if (value - previousValue[storeIndex] > 5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect" + Effect + "]", "next_effect", 1);
+                previousValue[storeIndex] = value;
+            } else if (value - previousValue[storeIndex] < -5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect" + Effect + "]", "prev_effect", 1);
+                previousValue[storeIndex] = value;
+            }
+        } else {
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FxOff = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
+        engine.setValue(group, "enabled", value ? 1 : 0);
+    }
+    // Nothing in single-effect mode
+};
+
+ReloopBeatmix24.EffectClearTimerCallBack = function(group) {
+    engine.setValue(group, "clear", 0);
+};
+
+ReloopBeatmix24.ShiftFxOff = function(channel, control, value, status, group) {
+    if (FxMode != 1) {
```

Comment:
this one, I don't understand why. FxMode is only set by this script and is always set to an integer, so what type mess could happen ?